### PR TITLE
API-6676: changes to allow code to compile with Scala 2.13. Some test…

### PR DESCRIPTION
**Pull request by worthydolt**
…s yet to fix

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,40 +1,36 @@
...
 import bloop.integrations.sbt.BloopDefaults
 
 lazy val appName = "api-definition"
+scalaVersion := "2.13.8"
 
 lazy val ComponentTest = config("component") extend Test
 
-ThisBuild / scalafixDependencies += "com.github.liancheng" %% "organize-imports" % "0.6.0"
-
+//ThisBuild / scalafixDependencies += "com.github.liancheng" %% "organize-imports" % "0.6.0"
```
Comment (AndySpaven) [inline build.sbt:12]: This is required for scalafix rules

Comment (worthydolt) [reply build.sbt]: Restored

File: app/uk/gov/hmrc/apidefinition/controllers/controllers.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ package object controllers {
 
   /** Used to improve the error messages that request.body.validate might return.
     */
-  def validationResult(errors: Seq[(JsPath, Seq[JsonValidationError])]): JsValue = {
+  def validationResult(errors: Seq[(JsPath, collection.Seq[JsonValidationError])]): JsValue = {
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apidefinition/controllers/controllers.scala:57]: I don't know what the way forward should be for these interactions with Play JSON but I have done things like this in other repos

`successful(UnprocessableEntity(validationResult(errors.map(error => (error._1, error._2.toSeq)).toSeq)))`

**Review by peteslater-ee — APPROVED** [review]

File: app/uk/gov/hmrc/apidefinition/controllers/controllers.scala [hunk 3]
```diff
@@ -45,7 +45,7 @@ package object controllers {
       {
         result =>
           result.fold(
-            errors => successful(UnprocessableEntity(validationResult(errors))),
+            errors => successful(UnprocessableEntity(validationResult(errors.toSeq))),
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apidefinition/controllers/controllers.scala:48]: I don't know what the way forward should be for these interactions with Play JSON but I have done things like this in other repos

`successful(UnprocessableEntity(validationResult(errors.map(error => (error._1, error._2.toSeq)).toSeq)))`

Comment (worthydolt) [reply app/uk/gov/hmrc/apidefinition/controllers/controllers.scala]: That's much nicer than the way that I tried to get rid of mutable `Seq`s. I've rolled your suggestions into this PR

File: app/uk/gov/hmrc/apidefinition/models/JsonFormatters.scala [hunk 4]
```diff
@@ -139,7 +140,7 @@ object JsonFormatters {
 
 object EnumJson {
 
-  def enumReads[E <: Enumeration](enum: E): Reads[E#Value] = new Reads[E#Value] {
+  def enumReads[E <: Enumeration](`enum`: E): Reads[E#Value] = new Reads[E#Value] {
```
[the remark this answers is not in the document]
Comment (worthydolt) [reply app/uk/gov/hmrc/apidefinition/models/JsonFormatters.scala]: Done the same

**Review by peteslater-ee — APPROVED** [review]

**Pull request merged**